health.go

```diff
@@ -46,7 +45,8 @@ func New(opts ...Option) Health {
 type health struct {
 	results        map[string]Result
 	checkTasks     map[string]checkTask
-	checksListener CheckListener
+	checksListener CheckListeners
+	healthListener HealthListeners
```

Comment:
I think these should be `CheckListener` and `HealthListener` respectively. `With*Listener` should check if the existing instance in the health struct is already a slice (ie `cl, ok := h.(CheckListeners)`) and append if so, create a slice otherwise.

That way the health struct remains unaware of the implementation.

Reply:
That will add unnecessary checks to the internal implementation. Simpler, cleaner and easier to follow code documents itself.

Reply:
I disagree. Currently an implementation is bound to the health struct instead of an interface. IMHO that's worse than a little complexity in a function that's sole purpose is to configure a listener.